### ISSUE TITLE
Indicator setup form with linked results: Rationale prompts appear when we are not logging the change or the rationale, with wrong success message appearing

### ISSUE THREAD

**Issue by jennyhavah** [mercycorps/TolaActivity#1412]
# To reproduce the issue

1. Log in as a user with High permissions.
2. Find an indicator that has linked results.
3. Open up the indicator setup form and make changes to any field _except for_ the Indicator name or anything on the Targets tab. (These are the types of changes we're logging, and they seem to be working as expected.)
4. Click "Save changes". Note that you are prompted for a rationale. 
    _This is the first thing that is unexpected, because you haven't changed anything we're supposed to be change logging._
5. Type something in the rationale field and click "OK".

Note that the success message is incorrect. Here's what you see -- the message you normally get immediately after creating an indicator:

![image](https://user-images.githubusercontent.com/33670923/56175302-7fd19a00-5fab-11e9-80a0-dd0eb018fa4a.png)

This is the one I would have expected -- copied from Production:

![image](https://user-images.githubusercontent.com/33670923/56175314-895b0200-5fab-11e9-98e0-a6ef907c8ff5.png)

6. Open up the change log for the program by clicking the link in the lower left corner under the IPTT table. Note that your changes were not logged, nor was your rationale. We don't actually _want_ to be logging these changes, but if we're going to prompt you to type something in, it's rude not to save it.

# Ideal behaviors
This is what Hanna and I would have _liked_ to see, with the understanding that we may need to settle for something less than perfect in the short term

- [ ] ~1. I am only prompted for a rationale in the following circumstances:~
     ~- Indicator has linked results~
     ~- I've modified the indicator name field on the Summary tab; and/or~
     ~- I've modified anything on the Targets tab, including: removing all targets , removing individual event targets, and adding one or more new event targets. (The event related stuff needs work that will be filed in a separate ticket -- just flagging it here.)~ _**Moved to #1427**_

- [x] 2. Upon submitting results after NOT getting prompted for a rationale, I see the expected success message, not the one you get after first creating an indicator.

## Potential alternative
If the above would be a big lift to implement, an acceptable (if sub-optimal) solution would be to leave the bajillion prompts in place and at least log the changes so you can see the rationales you took the trouble to enter.

**Comment by KennyMonster** [comment]
I ended up implementing the "Potential alternative" solution. The ideal solution is doable, but is a much heavier lift than I would like to do towards end of day before a desired release.

If nothing else comes up tomorrow I can work on it then.

**Comment by jennyhavah** [comment]
Thanks, Ken. I split the more desirable behavior into a separate ticket -- #1427 -- so that we can close this one as soon as it tests OK, without losing sight of what we really want.

**Comment by jennyhavah** [comment]
@KennyMonster I seem to be consistently getting the _wrong_ error message, whenever I edit a saved indicator. This one:

![image](https://user-images.githubusercontent.com/33670923/56255802-0d2fef80-607b-11e9-87a1-707d66791f19.png)

That is the message I expect when I've just added the indicator. The one we want is the second one displayed in the original ticket.

**Comment by KennyMonster** [comment]
I missed a spot. I think this should be good now.

**Comment by RickieT** [comment]
@KennyMonster was the solution to log everything? i was prompted for changing the "number" under the performance tab, and this is what the log showed:
![Screen Shot 2019-04-17 at 2 18 55 PM](https://user-images.githubusercontent.com/23220180/56321543-c47c4300-611b-11e9-8a5a-ef73b04560a2.png)

**Comment by KennyMonster** [comment]
@RickieT Correct. The rationale shows up, but the previous/new fields are blank since none of the "tracked" fields changed. We can add better UI checking in the future, just didn't want to do it while pushing for release.

**Comment by jennyhavah** [comment]
Hmm.. My "alternative" suggestion was to "at least log the changes so you can see the rationales you took the trouble to enter."... I guess I assumed that it was an "all or nothing" thing where we'd start showing not only the rationale text but also the actual changes.

I'm gathering from @KennyMonster 's comments that including the actual changes would be a significantly higher level of effort. Ultimately, we want to implement #1427.

So... Unfortunately, I don't think the current state of things is okay for release. Users are still going to be forced to enter a rationale whenever they edit indicators that have linked results. And the on top of that, we're not actually going to log the changes, so the rationales will be meaningless.

I think the next step is to estimate the level of effort for #1427. We have one more week before release, and so depending on the LoE for that ticket, we may want to go ahead and do it.

@hannalcamp let me know if you have a different take on this.

**Comment by jennyhavah** [comment]
@KennyMonster Any idea why the "Rationale for target" field is showing up when I make a non-tracked change and enter a rationale? ("Rationale for target" is a longstanding free-form text field that has nothing to do with the introduction of change logging and rationales for changes.)

![image](https://user-images.githubusercontent.com/33670923/56334624-e8f11300-614d-11e9-9aa9-b623973c5cf6.png)

**Comment by jennyhavah** [comment]
@hannalcamp @KennyMonster So, after chatting with both of you in Slack, this is where I'm leaning:

1. Ken reverts the "force save" flag he added, that is resulting in all new entries being saved to the log even if the changes are not to tracked fields. While I appreciate that it saves words users type, I think it's ultimately the wrong direction to go... It doesn't solve the problem of being prompted all the damn time about changes we are not actually tracking, and introduces a lot of confusion into the change log. We're going to see all sorts of rationales with no information about which fields they apply to and what actually changed... because they're not the fields we are actually tracking.

2. We really need to fix the "prompting you all the damn time" issue. I've updated #1427 to remove some complexity I didn't realize I was including (noted in the comments). Let's see how big a lift it is, in its current state.

**Comment by jennyhavah** [comment]
@KennyMonster Can I get some guidance on what to test at this point? Should I try changing a bunch of fields that are NOT tracked, to see if I get a prompt? Anything else?

**Comment by KennyMonster** [comment]
@jennyhavah My own testing has involved setting up an indicator with, and one without results. Then testing no changes, changing untracked, changing tracked. Then testing both of these on the modal version and full page version. Then various changes to target types. I look at the prompt, and also the audit log message.

As mentioned, there's a lot of permutations possible here, and I'm sure I missed some at this point.

**Comment by jennyhavah** [comment]
I tested the following:

Full page form without results
Modal form without results
Modal form with results

In all cases, I tried:
1. Saving changes without changing anything
2. Saving non-tracked changes
3. Saving tracked changes

In all cases, I was prompted when I expected to be prompted, and -- when prompted -- changes appeared with rationales in the log, as expected. 

In the case of deleting targets with no results, the deletion appeared in the log, with no rationale, as expected.
